openedx/core/djangoapps/content/course_overviews/models.py

```diff
@@ -0,0 +1,251 @@
+        from the given course.
+
+        Arguments:
+            course (CourseDescriptor): any course descriptor object
+
+        Returns:
+            CourseOverview: overview extracted from the given course
+        """
+        return CourseOverview(
+            id=course.id,
+            _location=course.location,
+            display_name=course.display_name,
+            display_number_with_default=course.display_number_with_default,
+            display_org_with_default=course.display_org_with_default,
+
+            start=course.start,
+            end=course.end,
+            advertised_start=course.advertised_start,
+
+            course_image_url=course_image_url(course),
+            facebook_url=course.facebook_url,
+            social_sharing_url=course.social_sharing_url,
+
+            certificates_display_behavior=course.certificates_display_behavior,
+            certificates_show_before_end=course.certificates_show_before_end,
+            has_active_web_certificates=(get_active_web_certificate(course) is not None),
+            cert_name_short=course.cert_name_short,
+            cert_name_long=course.cert_name_long,
+            lowest_passing_grade=course.lowest_passing_grade,
+            end_of_course_survey_url=course.end_of_course_survey_url,
+
+            mobile_available=course.mobile_available,
+            visible_to_staff_only=course.visible_to_staff_only,
+            _pre_requisite_courses_json=json.dumps(course.pre_requisite_courses)
+        )
+
+    @staticmethod
+    def get_from_id(course_id):
+        """
+        Load a CourseOverview object for a given course ID.
+
+        First, we try to load the CourseOverview from the database. If it
+        doesn't exist, we load the entire course from the modulestore, create a
+        CourseOverview object from it, and then cache it in the database for
+        future use.
+
+        Arguments:
+            course_id (CourseKey): the ID of the course overview to be loaded
+
+        Returns:
+            CourseOverview: overview of the requested course
+        """
+        course_overview = None
+        try:
+            course_overview = CourseOverview.objects.get(id=course_id)
+            # Cache hit! Just return the overview
+        except CourseOverview.DoesNotExist:
+            # Cache miss. Load entire course and create a CourseOverview from it
+            course = modulestore().get_course(course_id)
+            if course:
+                course_overview = CourseOverview._create_from_course(course)
+                course_overview.save()  # Save new overview to the cache
+        return course_overview
+
+    def __repr__(self):
+        """
+        Returns a simple string representation of this object for debugging.
+
+        Example return value: CourseOverview(location=course-v1://edX+DemoX.1+2014)
+        """
+        return "{}(location={})".format(self.__class__.__name__, self.location)
```

Comment:
nitpick: consider using `id` here instead of `location`.  Location is too verbose with the prefixed schema.